### PR TITLE
Add apt::keyring defined type

### PR DESCRIPTION
**Pull request by jps-help**
## Summary
Add a new defined type for adding GPG keyrings.

## Additional Context
apt-key is deprecated in the latest Debian and Ubuntu releases. Therefore we should move away from the old `apt::key` defined type which relies on it.
With this we can manage apt keyrings directly with the apt module. Further work would be required to integrate this directly into the `apt::source` like how `apt::key` is currently.
Reference - https://wiki.debian.org/DebianRepository/UseThirdParty

## Related Issues (if any)
#1034

## Checklist
- [x] 🟢 Spec tests - I don't have much experience with this, so I've simply generated the bare-minumum one with pdk.
- [ ] 🟢 Acceptance tests.
- [x] Manually verified. (For example `puppet apply`)

### CODE REVIEW

File: manifests/keyring.pp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#   Source of the keyring file. Mutually exclusive with 'content'.
+#
+# @param content
+#   Content of the keyring file. Mutually exclusive with 'source'.
+#
+# @param ensure
+#   Ensure presence or absence of the resource.
+#
+define apt::keyring (
+  Stdlib::Absolutepath $keyring_dir = '/etc/apt/keyrings',
```
Comment (elfranne) [inline manifests/keyring.pp:35]: This is already defined as `apt::params::trusted_gpg_d`

Comment (jamesps-ebi) [reply manifests/keyring.pp:35]: True, but for new-style APT keyrings, `/etc/apt/keyrings` is the preferred location.

Changing it only here means the new-style keyrings can be deployed to `/etc/apt/keyrings` by default, without affecting the rest of the module.

**Comment by kenyon** [comment]
Should be closed, was done in #1128.

**Comment by smortex** [comment]
Thanks @kenyon!

**Pull request closed without merging**